arkane/mainTest.py

```diff
@@ -0,0 +1,110 @@
+    """
+    Run all of Arkane's examples, and report which one failed
+    """
+    @classmethod
+    def setUp(self):
+        """A function that is run ONCE before all unit tests in this class."""
+        self.base_path = os.path.join(os.path.dirname(os.path.dirname(rmgpy.__file__)), 'examples', 'arkane')
+        self.failed = []
+        self.example_types = ['species', 'reactions', 'networks']
+
+    def test_arkane_examples(self):
+        for example_type in self.example_types:
+            example_type_path = os.path.join(self.base_path, example_type)
+            examples = []
+            for (dirpath, dirnames, filenames) in os.walk(example_type_path):
+                examples.extend(dirnames)
+                break
+            for example in examples:
+                path = os.path.join(example_type_path, example)
+                arkane = Arkane(inputFile=os.path.join(path, 'input.py'), outputDirectory=path)
+                arkane.plot = True
+                arkane.execute()
+                with open(os.path.join(path, 'arkane.log'), 'r') as f:
+                    log = f.readlines()
+                for line in log[::-1]:
+                    if 'execution terminated' in line:
+                        break
+                else:
+                    self.failed.append([example_type, example + 'FreeRotor'])
+        error_message = 'Arkane example(s) failed: '
+        for type_name_tuple in self.failed:
+            error_message += '{1} in {0}; '.format(type_name_tuple[0], type_name_tuple[1])
+        self.assertTrue(len(self.failed) == 0, error_message)
+
+    @classmethod
+    def tearDown(self):
+        """A function that is run ONCE after all unit tests in this class."""
+        self.extensions_to_delete = ['pdf', 'csv', 'txt', 'inp']
+        self.files_to_delete = ['arkane.log', 'output.py']
+        self.files_to_keep = ['README.txt']  # files to keep that have extentions marked for deletion
+        self.base_path = os.path.join(os.path.dirname(os.path.dirname(rmgpy.__file__)), 'examples', 'arkane')
+        self.example_types = ['species', 'reactions', 'networks']
+        for example_type in self.example_types:
+            example_type_path = os.path.join(self.base_path, example_type)
+            examples = []
+            for (dirpath, dirnames, filenames) in os.walk(example_type_path):
+                examples.extend(dirnames)
+                break
+            for example in examples:
+                # clean working folder from all previous test output
+                example_path = os.path.join(example_type_path, example)
+                dirs = [d for d in os.listdir(example_path)
+                        if not os.path.isfile(os.path.join(example_path, d))]
+                for d in dirs:
+                    shutil.rmtree(os.path.join(example_path, d, ''))
+                files = [f for f in os.listdir(example_path) if os.path.isfile(os.path.join(example_path, f))]
```

Comment:
It might be a bit cleaner to loop through the values of `os.listdir` and use an if statement to determine whether its a file or directory:
```
for name in os.listdir(example_path):
    if os.path.isfile:
        check filename and remove file
    else:
        remove directory
```